tst/testall.g

```diff
@@ -1,8 +1,17 @@
 LoadPackage( "cryst" );
-LoadPackage( "crystcat" );
 
-TestDirectory( DirectoriesPackageLibrary( "cryst", "tst" ),
-  rec( exitGAP     := true,
+alltests := DirectoriesPackageLibrary( "cryst", "tst" );;
+crystcat := List(["crystcat.tst"], f -> Filename(alltests,f));;
+
+if LoadPackage( "crystcat" ) = true then
+  TestDirectory( crystcat,
+    rec( exitGAP     := false,
+         testOptions := rec( compareFunction := "uptowhitespace") ) );
```

Comment:
This is not correct, as it ignores the result of the tests; the return value of `TestDirectory` would need to be honored.

But the better approach is to only call `TestDirectory` once, but change the exclusion list passed to it. Something like this (pseudo-code):
```gap
exclude := [];
if LoadPackage( "crystcat" ) = fail then
  exclude := ["crystcat.tst"];
fi;
TestDirectory( DirectoriesPackageLibrary( "cryst", "tst" ),
   rec( exclude := exclude, ...
```
Also, please don't use double semicolons needlessly.